pkg/sampler/sampler.go

```diff
@@ -33,15 +35,13 @@ func (t *Target) SetHash() {
 
 type Sample struct {
 	StatusCode      int
-	T1              time.Time
-	T2              time.Time
-	ResponseHeaders map[string]string
-}
-
-// Latency returns the amount of milliseconds between T1
-// and T2 (start and finish).
-func (s Sample) Latency() float64 {
-	return s.T2.Sub(s.T1).Seconds() * 1000
+	TimeStart       time.Time
+	TimeToConnect   time.Time
+	TimeToFirstByte time.Time
+	TimeEnd         time.Time
```

Comment:
+1 to downstream dealing with deltas. I suppose we could calculate the deltas here as well but meh.

Nit: it should probably be `ConnectTime` and not `TimeConnect`, ditto for `FirstByteTime` instead of `TimeFirstByte`.